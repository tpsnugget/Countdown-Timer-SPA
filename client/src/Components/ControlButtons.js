import React, { useState } from 'react'
import PropTypes from 'prop-types'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'

const ControlButtons = props => {

    const [oneXButton, setOneXButton] = useState(true)
    const [onePointFiveXButton, setOnePointFiveXButton] = useState(false)
    const [twoXButton, setTwoXButton] = useState(false)

    return (
        <Grid container justify='center' style={{marginTop: '3rem'}}>
            <Grid item xs={12} md={2} style={{marginBottom: '1rem'}}>
                {oneXButton ?
                    <Button data-testid='ControlButtons-oneX' variant="contained" color="primary" style={{width: '150px'}} >
                        <Typography variant='h4'>
                            1X
                        </Typography>
                    </Button>
                    :
                    <Button variant="contained" style={{width: '150px'}} >
                        <Typography variant='h4'>
                            1X
                        </Typography>
                    </Button>
                }
            </Grid>
            <Grid item xs={12} md={2} style={{marginBottom: '1rem'}}>
                {onePointFiveXButton ?
                    <Button variant="contained" color="primary" style={{width: '150px'}} >
                        <Typography variant='h4'>
                            1.5X
                        </Typography>
                    </Button>
                    :
                    <Button data-testid='ControlButtons-onePointFiveX' variant="contained" style={{width: '150px'}} >
                        <Typography variant='h4'>
                            1.5X
                        </Typography>
                    </Button>
                }
            </Grid>
            <Grid item xs={12} md={2}>
                {twoXButton ?
                    <Button variant="contained" color="primary" style={{width: '150px'}} >
                        <Typography variant='h4'>
                            2X
                        </Typography>
                    </Button>
                    :
                    <Button data-testid='ControlButtons-twoX' variant="contained" style={{width: '150px'}} >
                        <Typography variant='h4'>
                            2X
                        </Typography>
                    </Button>
                }
            </Grid>
        </Grid>
    )
}

ControlButtons.propTypes = {

}

export default ControlButtons
