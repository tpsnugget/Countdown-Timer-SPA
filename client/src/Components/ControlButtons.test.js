import React from 'react'
import { render } from '@testing-library/react'
import ControlButtons from './ControlButtons'

test('<ControlButtons />', () => {
    const { container, getByTestId } = render(<ControlButtons />)

    // Take a snapshot
    expect(container.firstChild).toMatchSnapshot()

    // Get the 1X button
    const oneXButton = getByTestId('ControlButtons-oneX')
    expect(oneXButton).toHaveProperty('type', 'button')

    // Get the 1.5X button
    const onePointFiveXButton = getByTestId('ControlButtons-onePointFiveX')
    expect(onePointFiveXButton).toHaveProperty('type', 'button')

    // Get the 2X button
    const twoXButton = getByTestId('ControlButtons-twoX')
    expect(twoXButton).toHaveProperty('type', 'button')

})