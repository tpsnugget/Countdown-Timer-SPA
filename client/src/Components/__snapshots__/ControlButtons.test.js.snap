// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlButtons /> 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  style="margin-top: 3rem;"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
    style="margin-bottom: 1rem;"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="ControlButtons-oneX"
      style="width: 150px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          1X
        </h4>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
    style="margin-bottom: 1rem;"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      data-testid="ControlButtons-onePointFiveX"
      style="width: 150px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          1.5X
        </h4>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      data-testid="ControlButtons-twoX"
      style="width: 150px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          2X
        </h4>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
